app/src/main/java/io/github/droidkaigi/confsched2017/view/activity/SessionDetailActivity.java

```diff
@@ -13,10 +13,18 @@
 public class SessionDetailActivity extends BaseActivity {
 
     private static final String EXTRA_SESSION_ID = "session_id";
+    private static final String EXTRA_PARENT = "parent";
 
-    public static Intent createIntent(@NonNull Context context, int sessionId) {
+    public static final int PARENT_MAIN = 0;
+    public static final int PARENT_MY_SESSIONS = 1;
+    public static final int PARENT_SEARCH = 2;
+
+    private int parent;
+
+    public static Intent createIntent(@NonNull Context context, int sessionId, int parent) {
```

Comment:
I think it's better to pass `Class` parameter instead of `int`. How about this? 😃 

```java
private Class parentClass;

public static Intent createIntent(@NonNull Context context, int sessionId, @Nullable Class clz) {
  Intent intent = new Intent(context, SessionDetailActivity.class);
  intent.putExtra(EXTRA_SESSION_ID, sessionId);
  if (clz != null) {
    intent.putExtra(EXTRA_PARENT_CLASS, clz.getName());
  }
}

@Override
protected void onCreate(Bundle savedInstanceState) {
  ...
  String clz = getIntent().getStringExtra(EXTRA_PARENT_CLASS, MainActivity.class.getName());
  parentClass = Class.forName(clz);
  ...
}

private void upToMainActivity() {
  Intent upIntent = new Intent(getApplicationContext(), parentClass);
  ...
}
```